docs/marketplace/dev/front-end/table-design/index.md

```diff
@@ -0,0 +1,361 @@
+
+### Features
+
+Every other piece of functionality was extracted into so called `Table Feature`:
+
+- Table Feature is simply an Angular Component that encapsulates specific extension of the Core Table.
+- Core Table contains specific placeholders in it’s view that Table Feature may target to render it’s own piece of UI.
+- Most of the common table functionality already exists as a Table Feature and may be used in project.
+
+To use Features, Table Module must implement a specific interface (TableConfig) and then be registered to the Root Module via TableModule.withFeatures():
+
+```ts
+class TableDefaultConfigData implements Partial<TableConfig> {
+    total = {
+        enabled: true,
+    };
+    columnConfigurator = {
+        enabled: true,
+    };
+}
+
+@NgModule({
+    imports: [
+        TableModule.forRoot(),
+        TableModule.withFeatures({
+            filters: () => import('@spryker/table.feature.filters').then((m) => m.TableFiltersFeatureModule),
+            pagination: () => import('@spryker/table.feature.pagination').then((m) => m.TablePaginationFeatureModule),
+            rowActions: () => import('@spryker/table.feature.row-actions').then((m) => m.TableRowActionsFeatureModule),
+            search: () => import('@spryker/table.feature.search').then((m) => m.TableSearchFeatureModule),
+            syncStateUrl: () => import('@spryker/table.feature.sync-state').then((m) => m.TableSyncStateFeatureModule),
+            total: () => import('@spryker/table.feature.total').then((m) => m.TableTotalFeatureModule),
+            itemSelection: () =>
+                import('@spryker/table.feature.selectable').then((m) => m.TableSelectableFeatureModule),
+            batchActions: () =>
+                import('@spryker/table.feature.batch-actions').then((m) => m.TableBatchActionsFeatureModule),
+            columnConfigurator: () =>
+                import('@spryker/table.feature.settings').then((m) => m.TableSettingsFeatureModule),
+            title: () => import('@spryker/table.feature.title').then((m) => m.TableTitleFeatureModule),
+            editable: () => import('@spryker/table.feature.editable').then((m) => m.TableEditableFeatureModule),
+        }),
+    ],
+    providers: [
+        {
+            provide: TableDefaultConfig,
+            useClass: TableDefaultConfigData,
+        },
+    ],
+})
+export class DefaultTableConfigModule {}
+```
+
+### Columns
+
+Columns in a Table are defined by the `Column Type` and are rendered withing the columns (text, image, link, etc.).
+New Column Type may be created and registered to the Table.
+
+To use Column components, Table Module must implement a specific interface (TableConfig) and then be registered to the Root Module via TableModule.withColumnComponents():
+
+```ts
+class TableDefaultConfigData implements Partial<TableConfig> {
```

Comment:
Please only keep example of how one column is registered and remove everything else.